### PR TITLE
Redirect Device Pages If Capitalization Isn't Right

### PR DESCRIPTION
**Pull request by danielcliu-ifixit**
## Overview
This got super easy after #556 
We want to now 1.) handle iFixit only devices (ghost devices) redirect to the correct capitalization 2.) prioritize iFixit's capitalization over Strapi's. This will make it easier to link from iFixit to the store so they don't have to check strapi for the official url, etc.

## QA
While testing watch out for permanent redirecting (308s and the like). You may have to clear your browsers cache (or in the code change the `[...deviceHandleItemType].tsx` redirect calls permanent attribute to false)
You should test an device not in strapi (`/Phone` is what I use) and make sure that even if you mess around with the capitalization it reroutes back to the correct URL. Also make sure that iFixit data is prioritized over strapi's, so maybe change some local strapi data to be miscapitalized and then make sure the iFixit data is used instead.

Closes: #571

### CODE REVIEW

File: frontend/models/product-list/index.ts [hunk 1]
```diff
@@ -61,12 +61,11 @@ export async function findProductList(
    ]);
 
    const productList = result.productLists?.data?.[0]?.attributes;
-
    if (productList == null && deviceWiki == null) {
       return null;
    }
 
-   const deviceTitle = productList?.deviceTitle ?? filterDeviceTitle;
+   const deviceTitle = deviceWiki?.deviceTitle ?? productList?.deviceTitle;
```
Comment (davidrans) [inline frontend/models/product-list/index.ts:68]: Does this need to be a string?
```suggestion
   const deviceTitle = deviceWiki?.deviceTitle ?? productList?.deviceTitle ?? '';
```

What was `filterDeviceTitle`?

Comment (masonmcelvain) [reply frontend/models/product-list/index.ts:68]: `filterDeviceTitle` is taken directly from the url. So for `/Parts/ibook` it would be `ibook`.

Comment (masonmcelvain) [reply frontend/models/product-list/index.ts:68]: We should probably default to string or null here: https://github.com/iFixit/react-commerce/blob/4b59f9c916a424b5e38b49f9a93781e4364ccf02/frontend/models/product-list/types.ts#L50

Comment (sterlinghirsh) [inline frontend/models/product-list/index.ts:68]:
```suggestion
   const deviceTitle = deviceWiki?.deviceTitle ?? productList?.deviceTitle ?? null;
```

Comment (sterlinghirsh) [reply frontend/models/product-list/index.ts]: Good call. I made it null.

**Comment by sterlinghirsh** [comment]
CR :zap:

**Review by sterlinghirsh — APPROVED** [review]

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻

File: frontend/models/product-list/index.ts (hunk 1 above)
Comment (danielcliu-ifixit) [reply frontend/models/product-list/index.ts]: This is cleaner so I'm not complaining but I think the previous version still worked

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻

**Comment by rickisXP** [comment]
QA :writing_hand: but deploy_block :thought_balloon: on a potential issue
- Wonky capitalization gets casted to the correct capitalization :heavy_check_mark: 
- Changing Strapi `deviceTitle` to be miscapitalized will use iFixit data instead :heavy_check_mark: 
- Navigating to [`/pArTs`](https://react-commerce-prod-git-redirect-if-strapi-device-59a5ab-ifixit.vercel.app/pArTs) or [`/tOOls`](https://react-commerce-prod-git-redirect-if-strapi-device-59a5ab-ifixit.vercel.app/tOOls) will 404 :warning:

**Comment by sterlinghirsh** [comment]
un_deploy_block :zap: that's fine, since the parts/tools segment is outside the scope of this pr

**Comment by danielcliu-ifixit** [comment]
@rickisXP I should have clarified the scope of the change, it doesn't fix capitalization for `/Tools` or `/Parts` i think that change is coming with next js is updated. 
un_deploy_block :v:

**Pull request merged**